src/graph.py

```diff
@@ -1,242 +1,192 @@
-        """Add a new edge to the graph connecting 'n1' and 'n2'; if either n1 or n2 are not lready present in the graph, they should be added."""
-        self.is_node(node1)
-        self.is_node(node2)
-        self.set_of_nodes.add(node1)
-        self.set_of_nodes.add(node2)
-        node1.edges.add(node2)
-
-    def del_edge(self, node1, node2):
-        """Delete the edge connecting 'n1' and 'n2' from the graph, raises a Key Error if no such edge exists."""
+    def has_node(self, value):
+        return value in self.set_of_nodes
+
+
+    def add_edge(self, value1, value2, weight):
+        # import pdb; pdb.set_trace()
+        if not self.has_node(value1):
+            self.add_node(value1)
+        if not self.has_node(value2):
+            self.add_node(value2)
+        self.gnodes[value1][value2] = weight
+
+
+    def del_edge(self, value1, value2):
         try:
-            node1.edges.remove(node2)
+            del(self.gnodes[value1][value2])
         except KeyError:
             raise KeyError('Edge that you are trying to exist does not exist.')
-        except AttributeError:
-            raise TypeError('{} is not a GNode type.'.format(str(node1)))
 
-    def del_node(self, node_to_del):
-        """Delete a node from the graph, raises an error if no such node exists."""
+
+    def del_node(self, value):
         try:
-            self.set_of_nodes.remove(node_to_del)
+            self.set_of_nodes.remove(value)
         except KeyError:
             raise KeyError('Node does not exist in graph.')
-
-        for node in self.set_of_nodes:
+        del(self.gnodes[value])
+        for node in self.gnodes:
             try:
-                node.edges.remove(node_to_del)
+                del(self.gnodes[node][value])
             except KeyError:
                 pass
 
     def nodes(self):
-        """Return a list of nodes."""
         return list(self.set_of_nodes)
 
+
     def edges(self):
-        """Return a list of edges."""
-        list_of_edges = []
-        for node in self.set_of_nodes:
-            for edge in node.edges:
-                list_of_edges.append(Edge(node, edge))
-        return list_of_edges
+        edge_list = []
+        for node in self.gnodes:
+            for key in self.gnodes[node]:
+                edge = '{0}->{1}'.format(node, key)
+                edge_list.append(edge)
+        return edge_list
+
 
     def adjacent(self, node1, node2):
-        """Return true if two nodes are connected by an edge, false if not connected.  Will raise a TypeError is one node is not in the graph."""
-        self.is_node(node1)
-        self.is_node(node2)
-        if not self.has_node(node1) or not self.has_node(node2):
-            raise KeyError('Cannot check adjacent on a node that is not in the graph.')
-        else:
-            return node2 in node1.edges
+        if not(node1 in self.set_of_nodes and node2 in self.set_of_nodes):
+            raise TypeError('A node is not in the graph.')
```

Comment:
This is not the right error for this scenario -- why not just use KeyError like your other scenarios?